<h1><center>Capstone 2: narrative analytics and experimentation</center></h1>

The main component of this capstone is a research proposal. Using the data set you selected, propose and outline an experiment plan. The plan should consist of three key components:

* Analysis that highlights your experimental hypothesis.

* A rollout plan showing how you would implement and rollout the experiment

* An evaluation plan showing what constitutes success in this experiment

_________________________________________________________________________________________________________________

##### Experiment Rollout Plan

Randomly assign a roughly equal group of individuals to two groups, a treatment group and a control group. We will then be measuring whether individuals in the two groups will be converted to a customer status or not. We will then take the average comparison rate of each group to see which group experiences a more productive page. In order to ensure that we are finding whether there is in fact a statisitcal significance in the difference between the two groups, we will be evaluating the t-statistic and p-value to ensure that there is in fact a statistically significant differnce between the means and the difference is not due to jsut noise.

We will then, after evaluating the effectiveness of the test, determine which web page to rollout to all users.

In [15]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

In [3]:
experiment_data = pd.read_csv('ab_data.csv')

In [4]:
experiment_data.head()

user_id                   timestamp      group landing_page  converted
0   851104  2017-01-21 22:11:48.556739    control     old_page          0
1   804228  2017-01-12 08:01:45.159739    control     old_page          0
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0
4   864975  2017-01-21 01:52:26.210827    control     old_page          1

In [5]:
experiment_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 294478 entries, 0 to 294477
Data columns (total 5 columns):
user_id         294478 non-null int64
timestamp       294478 non-null object
group           294478 non-null object
landing_page    294478 non-null object
converted       294478 non-null int64
dtypes: int64(2), object(3)
memory usage: 11.2+ MB


### Experimental Hypothesis

The null hypothesis is that there will be no effect from the trial. Therefore our null hypothesis is that the treatment group and the control group will experience the same conversation rate.

Lets find below what our conversation rate is for the treatment and control group.

In [13]:
# How large is each group?
print('test sample size:', len(experiment_data[experiment_data.group == 'treatment']))
print('control sample size:', len(experiment_data[experiment_data.group == 'control']))

test sample size: 147276
control sample size: 147202


In [12]:
# What is the signup rate of test and control?
print('test signup rate:', experiment_data[experiment_data.group == 'treatment'].converted.mean())
print('control signup rate:', experiment_data[experiment_data.group == 'control'].converted.mean())

test signup rate: 0.11891957956489856
control signup rate: 0.12039917935897611


Both group sizes are rather large, and also have very similar values. We can then anticipate that the signup rates will be reliable in understanding the effect on conversion and not just due to noise.

We can see that the test signup rate is only %11.9 and the control signup rate is %12

In [16]:
# Is the experiment already significant?
stats.ttest_ind(
  experiment_data[experiment_data.group == 'treatment'].converted,
  experiment_data[experiment_data.group == 'control'].converted)

Ttest_indResult(statistic=-1.2369207675839395, pvalue=0.21611748562490837)

We have a very small t-value, indicating that the difference in means is not significant. This is reinforced by a p-value of 0.22, confirms that the means are not different and that we can accept the null hypothesis of no effect.

##### Was the new page a success?

The test signup rate and the control signup rate was almost exactly equal. Additionally, we see that the t-value was low and the p-value was greater than 0.05. We therefore can state that there was no difference between the conversion rates of the two sites. Therefore, I would pick whichever site is the cheapest to operate.